File: src/main/java/services/MessageService.java
package services;

import database.Database;
import models.Message;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class MessageService {

    private Map<Long, Message> messages = Database.getMessages();

    public MessageService() {
        messages.put(1L, new Message(1, "This is the first message ever.", "Hassan"));
        messages.put(2L, new Message(2, "This is the second message.", "Ward"));
    }

    public List<Message> getMessages() {
        return new ArrayList<>(messages.values());
    }

    public Message getMessage(Long id) {
        return messages.get(id);
    }

    public Message addMessage(Message message) {
        message.setId(messages.size() + 1);
        messages.put(message.getId(), message);
        return message;
    }

    public Message updateMessage(Message message) {
        if (message.getId() <= 0) return null;
        messages.put(message.getId(), message);
        return message;
    }

    public Message removeMessage(long id) {
        return messages.remove(id);
    }
}

File: src/main/java/services/MessageService.kt
package services

import database.Database
import models.Message
import java.util.ArrayList

class MessageService() {

    var messages: HashMap<Long, Message> = Database.messages

    init {
        messages[1L] = Message(id = 1L, message = "Hello Zere!", author = "Zirgovic")
    }

    fun getMessages(): List<Message> {
        return ArrayList<Message>(messages.values)
    }

    fun getMessage(id: Long): Message? {
        return messages[id]
    }

    fun addMessage(message: Message): Message? {
        message.id = messages.size + 1L
        messages[message.id] = message
        return message
    }

    fun updateMessage(message: Message): Message? {
        if (message.id < 0) return null
        messages[message.id] = message
        return message
    }

    fun deleteMessage(id: Long): Message? {
        println("delete ${messages.remove(id)}")
        return messages.remove(id)
    }

    fun printMessages() {
        for (m in messages)
            println(m.toString())
    }


}